drivers/dac/dac_sam0.c

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2020 Google LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT atmel_sam0_dac
+
+#include <errno.h>
+
+#include <drivers/dac.h>
+#include <soc.h>
+
+/*
+ * Maps between the DTS reference property names and register values
+ * TODO(#21273): replace once improved support for enum values lands.
+ */
+#define SAM0_DAC_REFSEL_0 DAC_CTRLB_REFSEL_INT1V_Val
+#define SAM0_DAC_REFSEL_1 DAC_CTRLB_REFSEL_AVCC_Val
+#define SAM0_DAC_REFSEL_2 DAC_CTRLB_REFSEL_VREFP_Val
```

Comment:
I suggest to use the same naming as in the [D21 datasheet](https://ww1.microchip.com/downloads/en/DeviceDoc/SAM_D21_DA1_Family_DataSheet_DS40001882F.pdf)?
- `INTREF`
- `VDDANA` (same as the pin name)
- `VREFA` (`VREFP` seems to be an error/typo in the [D20 datasheet](http://ww1.microchip.com/downloads/en/DeviceDoc/SAM_D20_%20Family_Datasheet_DS60001504C.pdf))

Reply:
Interesting, I've been using the Atmel 09/2015 revision.  Looks like these were renamed between that and the Microchip 03/2020 revision.

Changed to match the datasheet.

Reply:
Seems like this change didn't make it to the commit. It's still the old names (or github is displaying me the older version).

Reply:
That's correct.  The ASF used with Zephyr uses the old names, so I updated the DeviceTree binding and added a comment but didn't change the names.

Reply:
See the new comment:

```
/*
 * Maps between the DTS reference property names and register values.  Note that
 * the ASF uses the 09/2015 names which differ from the 03/2020 datasheet.
 *
 * TODO(#21273): replace once improved support for enum values lands.
```

Reply:
Ah, didn't realize that those defines come from the ASF. Thanks for adding a comment.